custom/icds_reports/management/commands/fix_past_data.py

```diff
@@ -0,0 +1,55 @@
+import os
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+
+from django.core.management.base import BaseCommand
+from corehq.util.argparse_types import date_type
+from django.db import connections, transaction
+from custom.icds_reports.tasks import _get_monthly_dates
+from corehq.sql_db.connections import get_icds_ucr_citus_db_alias
+
+
+@transaction.atomic
+def _run_custom_sql_script(command, day=None):
+    db_alias = get_icds_ucr_citus_db_alias()
+    if not db_alias:
+        return
+    with connections[db_alias].cursor() as cursor:
+        cursor.execute(command, [day])
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'month',
+            type=date_type,
+            help='The start date (inclusive). format YYYY-MM-DD'
+        )
+
+    def build_data(self, date):
+        for i in range(0, 7):
+            path = os.path.join(os.path.dirname(__file__), 'sql_scripts',
+                                'fix_past_data_part_{}.sql'.format(i))
+            end_date = date + relativedelta(months=1)
+            context = {
+                'start_date': date.strftime("%Y-%m-%d"),
+                'end_date': end_date.strftime("%Y-%m-%d")
+            }
+            with open(path, "r", encoding='utf-8') as sql_file:
+                sql_to_execute = sql_file.read()
+                sql_to_execute = sql_to_execute % context
```

Comment:
have you tested that this works the way you expect? I have never seen reading a string like this from a file and then formatting it after the fact. i thought formatting had to happen at string creation

Reply:
I think what you actually want to do is pass these as params to the sql query

Reply:
Yeah, it's done to pass the params to the sql query and it works correctly, in fact I am not sure about this initially when I tried in the cas-data-gen, https://github.com/dimagi/cas-data-gen/blob/master/data_gen/data/birth_preparedness.py Here is the sample where I have used the same.

Reply:
I think it would still be better to pass them as true parameters. using `cursor.execute(query, params)` rather than formatting them ourselves. that lets the underlying library handle the paraterization, and is the more standard way to pass arguments to a sql script

Reply:
If you are going to do it this way, please use standard string formatting `string.format(...)` rather than the old style `%` formatting